File: integration-prototypes/csharp/kafka-ha/tests/KafkaHaClients.Tests/KafkaHaTests.cs
using KafkaHaClients;

namespace KafkaHaClients.Tests;

public sealed class KafkaHaTests
{
    [Fact]
    public void BuildMessage_FormatsExpectedJson()
    {
        Assert.Equal("{\"id\":\"1\",\"text\":\"hello\"}", ProducerClient.BuildMessage("1", "hello"));
    }
}

File: integration-prototypes/java/kafka-ha/src/test/java/org/prototype/kafka/ha/KafkaHaTests.java
package org.prototype.kafka.ha;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

public final class KafkaHaTests {
    @Test
    void buildMessageFormatsJson() {
        assertEquals("{\"id\":\"1\",\"text\":\"hello\"}", ProducerClient.buildMessage("1", "hello"));
    }
}
